File: src/rb/results.rb
module Results
    class ParseSideResult
        def initialize(value)
            if value.is_a? Integer
                @success = true
                @side = value
            else
                @success = false
                @errorMessage = value
            end
        end

        def success
            @success
        end
        
        def side
            @side
        end
        
        def errorMessage
            @errorMessage
        end
    end

    class MatrixResult
        def initialize(matrix, sum, side, success)
            @matrix = matrix
            @sum = sum
            @side = side
            @success = success
        end

        def matrix
            @matrix
        end

        def sum
            @sum
        end

        def side
            @side
        end

        def success
            @success
        end
    end
end

File: src/cs/results.cs
namespace Results
{
    public class ParseSideResult
    {
        private ParseSideResult(bool success)
        {
            Success = success;
            ErrorMessage = string.Empty;
        }

        public ParseSideResult(int side) : this(true)
        {
            Side = side;
        }

        public ParseSideResult(string errorMessage) : this(false)
        {
            ErrorMessage = errorMessage;
        }
        
        public bool Success { get; }
        public int Side { get; }
        public string ErrorMessage { get; }
    }

    public class MatrixResult
    {
        public MatrixResult(int[,] matrix, int sum, int side, bool success)
        {
            Matrix = matrix;
            Sum = sum;
            Side = side;
            Success = success;
        }

        public int[,] Matrix { get; }
        public int Sum { get; }
        public int Side { get; }
        public bool Success { get; }
    }
}
